pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/authenticator/KerberosAuthenticator.java

```diff
@@ -0,0 +1,84 @@
+package org.pac4j.kerberos.credentials.authenticator;
+
+import java.util.HashMap;
+
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+import org.pac4j.kerberos.profile.KerberosProfile;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileHelper;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.profile.creator.AuthenticatorProfileCreator;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.InitializableObject;
+import org.pac4j.core.util.InitializableWebObject;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.authenticator.Authenticator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Authenticator for Kerberos. It creates the user profile and stores it in the credentials
+ * for the {@link AuthenticatorProfileCreator}.
+ *
+ * @author Garry Boyce
+ * @since 1.9.1
+ */
+public class KerberosAuthenticator extends InitializableWebObject implements Authenticator<KerberosCredentials> {
+
+    protected final Logger                logger = LoggerFactory.getLogger(getClass());
+
+    private KerberosTicketValidator ticketValidator;
+
+    public KerberosAuthenticator() {
+	}
+
+	/**
+     * Initializes the authenticator that will validate Kerberos tickets.
+     *
+     * @param ticketValidator    The ticket validator used to validate the Kerberos ticket.
+     * @since 1.9.1
+     */
+    public KerberosAuthenticator(KerberosTicketValidator ticketValidator) {
+        this.ticketValidator = ticketValidator;
+    }
+
+    @Override
+    public void validate(KerberosCredentials credentials) {
+        logger.debug("Try to validate Kerberos Token:" + credentials.getKerberosTicketAsString());
+        KerberosTicketValidation ticketValidation = this.ticketValidator.validateTicket(credentials.getKerberosTicket());
+        logger.debug("Kerberos Token validated");
+
+        String subject = ticketValidation.username();
+        logger.debug("Succesfully validated " + subject);
+
+        if (!subject.contains(UserProfile.SEPARATOR)) {
+            subject = KerberosProfile.class.getSimpleName() + UserProfile.SEPARATOR + subject;
+        }
+
+        CommonProfile profile = null;
+        try {
+            profile = ProfileHelper.buildUserProfileByClassCompleteName(subject, new HashMap<String, Object>(), KerberosProfile.class.getName());

```

Comment:
If I want to create an external security plugin as I did with kerberos before this iteration then I would want to do it in the same way was the other plugins so I really think this method should be public to give users the flexibility.